operator/templates/stateful-set.yaml

```diff
@@ -356,6 +364,11 @@ spec:
             name:  {{ $.Name }}-generate-tls-artifacts-sh
             defaultMode: 0755
         {{ end }}
+        {{ if $.Params.AUTHENTICATION_CREDENTIALS_SECRET }}
+        - name: {{ $.Params.AUTHENTICATION_CREDENTIALS_SECRET }}
```

Comment:
The name should probably not be the param, right? I'm not sure why it's used that way for the TLS_SECRET above, but the name here is just a fixed string that is used to reference the secret in the volumemounts above? I would prefer a fixed string here, otherwise people might try to set `TLS_SECRET_NAME` and `AUTHENTICATION_CREDENTIALS_SECRET` to the same secret and we would have duplicate entry in here.